connector/blobuploadconnector/logs.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package blobuploadconnector
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type passThroughLogsConnector struct {
```

Comment:
is this just meant to be a placeholder for the eventual implementation?  Consider using the name we will eventually use (e.g. logBlobUploader), to minimize future changes.

Reply:
It is not a placeholder in that the full implementation will still include the `passthrough` variant.

You can see more clearly where this is going in the more fleshed out implementation sketch in:

 - [`[michaelsafyan/otel-contrib] connector/blobuploadconnector/traces.go`](https://github.com/michaelsafyan/open-telemetry.opentelemetry-collector-contrib/blob/blob_upload_connector/connector/blobuploadconnector/traces.go)

In particular:

```
func createTracesToTracesConnectorWithRegistryFactory(...) {
  ...
  if cfg.Traces == nil {
	  ...
	  return &passThroughTracesConnector{
		  nextConsumer: nextConsumer,
	  }, nil
  }
  ...
 result := &tracesToTracesImpl{ ... }
 ...
 return result, nil
}
```

Basically, the real implementation has some overhead even when there are no rules in effect, and so the factory function will instantiate a pass-through rather than a real implementation if the config does not have any rules which would be applied. This can be the case, for example, when configuration applies only to traces but not logs.

Reply:
You don't need to do this. If someone doesn't want this processor, they should omit it from their trace pipeline, rather than include it without configuration.  You should return an error if someone creates a blobuploader for traces without including trace configuration.

Reply:
Seems reasonable.

I've renamed these so that they are placeholders for the eventual, real implementation.

When I build on this to upstream the other pieces of the implementation, I will be stricter in the validation rather than attempting to fail more gracefully.